### ISSUE TITLE
Low-end TV's freeze at multi-period joins during a DASH DRM Livestream

### ISSUE THREAD

**Issue by dstreet26** [shaka-project/shaka-player#4465]
**Have you read the [Tutorials](http://shaka-player-demo.appspot.com/docs/api/tutorial-welcome.html)?**
Yes

**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?**
Yes

**What version of Shaka Player are you using?**
4.2.0

**Please ask your question**

We have been continually running into an issue on low end TV's (specifically 2016-2018 era VIZIO's) on a multi-period DASH DRM livestream. The issue is that at the multi-period joins, those TV's will sometimes (not always, but a lot of time) freeze up - for upwards of a minute or more. We've repeated this issue with the shaka demo page, so it's probably not an issue with our web app. We only see this freezing on low power devices, such as low end VIZIO's. We have not seen it happening consistently on Tizen, in browser, etc.

We have a 11 minute portion of the livestream where it froze and saved (with MediaPackage's live to VOD clip harvest functionality) it to a separate video file to test over and over. (There is one period join at 540 seconds (9 minutes) and another period join at 605 seconds (10.3 minutes)).

**MPD file:** https://static.drm.pbs.org/harvestedclips/dash-drm-clip-7-29-130pm/manifest.mpd
**DRM license lookup URL:** _Will send separately to shaka-player-issues@google.com_ 

In the latest version of Shaka, we've noted the more verbose debugging logs (thanks! ❤️ ), and have noticed some consistent logs happening at the points where these freezes occur. We noticed that there are shaka logs/errors about: 

```
Possible encoding problem detected! Unexpected buffered range for reference
Jumping forward 0.42531800000006115 seconds because of gap starting at 605.238899 and ending at 605.338733
```

As well as:

`Stall detected at 731.664822 for 1.7569999694824219 seconds.`

Here is a video showing the Shaka debug logs while a clip of the livestream plays. You can see many occurrences of the TV freezing:
 
https://user-images.githubusercontent.com/6311278/187938324-fde34b92-2485-4e20-b611-c9f27d29da09.mp4

Here is a static screenshot of the errors:

<img width="1553" alt="Screen Shot 2022-08-31 at 3 31 43 PM" src="https://user-images.githubusercontent.com/6311278/187938377-e93d496f-2730-4aa9-94b5-82f6268a94e2.png">

Questions:

- What do these mean?
- Is jumping the same as making the player seek? Or is it just an adjustment to the underlying timing state/model?
- Are these indicating issues in the stream itself? Note - this stream is being generated by AWS MediaLive / MediaPackage.
- **Any general guidance on how to make mutiperiod dash streams behave on low end devices?**

Other notes:

We also noticed in the source code that there are checks for running on some devices (like Tizen) but not specifically for VIZIO.

(source code for isTizen: https://github.com/shaka-project/shaka-player/blob/main/lib/util/platform.js#L105)

So we hacked the shaka-player source code to force `isTizen()` to return **true** but that didn't seem to fix the issue on VIZIO.

Secondary question:

Can you also shed some light on what some of the different stuff that happens on Tizen is? Like:

`streaming.stallSkip = 0;`

And:

`streaming.gapDetectionThreshold = 0.5;`

Thanks in advance for any assistance you can give us!

**Comment by dilukd** [comment]
@dstreet26  - do you support Chromecast and have you noticed similar issues on it (on Chromecast Ultra or below mainly) ? we've been seeing something very similar at period boundaries.

**Comment by dstreet26** [comment]
We do support Chromecast in a separate project. It is a custom receiver.

Well I misplaced my Chromecast Ultra and I can no longer inspect a Gen 1 Chromecast due to a blank white screen, but I can still cast to a Gen 1 and my Gen 2.

Here is how I am enabling debugging on the Chromecast Receiver:

```
receiverContext.setLoggerLevel(cast.framework.LoggerLevel.DEBUG)
```

If I enable debugging, I DO NOT see the shaka "gap" events or many other Shaka events...

I'm not sure how to tell the Cast SDK's built-in shaka-player to debug more... If anybody knows let me know.

I tested the DRM livestream on a Gen1 and Gen2 device. There was no noticeable stuttering for over an hour.

The harvested clip also played fine.
